lua/autorun/server/bounties.lua

```diff
@@ -0,0 +1,89 @@
+util.AddNetworkString("Bounties")
+util.AddNetworkString("IncreaseBounty")
+
+hook.Add("PlayerDeath","PlayerDeath",function(ply,infl,atk)
+	local bounty = tonumber(ply:GetPData("bounty",0))
+	if bounty > 0 and ply != atk then
+		atk:PS_GivePoints(bounty)
+		BotSayGlobal("[edgy]"..atk:Nick().." [fbc]has claimed [gold]"..ply:Nick().."'s [fbc]bounty of [rainbow]"..bounty.." [fbc]points!")
+		ply:SetPData("bounty",0)
+	end
+end)
+
+function FindPlayer(plyname)
+	if type(plyname) == "string" and plyname != "" then
+		local search = {}
+		for k,v in pairs(player.GetAll()) do
+			local x,y,z = string.find(string.lower(v:GetName()),".*("..string.lower(plyname)..").*")
+			if z then
+				table.insert(search,v)
+			end
+		end
+		if table.Count(search) == 1 then
+			return search[1]
+		elseif table.Count(search) > 1 then
+			return "Multiple players found, be more specific."
+		elseif table.Count(search) == 0 then
+			return "No players found."
+		end
+		return "Error"
+	end
+end
+
+hook.Add("PlayerSay","BountyChatCommands",function(ply,text)
```

Comment:
Can you set bounties with either a chat command or a derma window? Please do just one or the other

Reply:
why can't it be both

Reply:
Hard to debug/maintain. If you still want to, please make a server side function AddBountyToPlayer(calling_ply, target_ply, amount) and handle as much of the logic as possible within it so there is minimal duplicate code

Reply:
I kind of need the derma window so that people can see who all has a bounty

Reply:
I could just get rid of the whole NewBounty function but the list with the derma is all pretty necessary

Reply:
You could make a chatcommand to print the list of people with bounties (only those with >0 bounty)